components/observations/ObservationsFilterForm.tsx

```diff
@@ -72,20 +103,11 @@ export const datesForFilterConfig = (dates: z.infer<typeof observationFilterConf
 };
 
 const matchesDates = (currentDate: Date, dates: z.infer<typeof observationFilterConfigSchema.shape.dates>): FilterFunction => {
-  let matches = true;
   if (!dates) {
     return () => true;
   }
   const {startDate, endDate} = datesForFilterConfig(dates, currentDate);
-  return (observation: ObservationFragment) => {
-    if (dates.from) {
-      matches = matches && isAfter(parseISO(observation.createdAt), startDate);
-    }
-    if (dates.to) {
-      matches = matches && isBefore(parseISO(observation.createdAt), endDate);
-    }
-    return matches;
-  };
+  return (observation: ObservationFragment) => isAfter(parseISO(observation.createdAt), startDate) && isBefore(parseISO(observation.createdAt), endDate);
```

Comment:
Not something you need to change, but I would like to mention that the moment you need to debug any of this or add a logging line, the one-liner starts to be something to shake your fist at :)

Reply:
well, the old version was actually incorrect (dates.to/dates.from aren't always set) so this is an improvement in that sense 😂 

prettier is the thing that aggressively turns these into one-liners, which I do have mixed feelings about

Reply:
Didn't the previous checks (` if (dates.from) {`) handle the unset case?

Reply:
no, the logic around those fields changed. `dates.from` / `dates.to` are now only set for custom date ranges. `datesForFilterConfig` wraps up the logic of extracting them as needed.

https://github.com/stevekuznetsov/avalanche-forecast/blob/0ed7d41314b81a2edbcae4f9786222bbd601f72f/components/observations/ObservationsFilterForm.tsx#L109-L110